daemon/algod/api/server/v2/handlers.go

```diff
@@ -1174,25 +1173,30 @@ func (v2 *Handlers) GetApplicationBoxes(ctx echo.Context, applicationID uint64,
 }
 
 // GetApplicationBoxByName returns the value of an application's box
-// (GET /v2/applications/{application-id}/boxes/{box-name})
-func (v2 *Handlers) GetApplicationBoxByName(ctx echo.Context, applicationID uint64, boxName string) error {
+// (GET /v2/applications/{application-id}/box)
+func (v2 *Handlers) GetApplicationBoxByName(ctx echo.Context, applicationID uint64, params generated.GetApplicationBoxByNameParams) error {
 	appIdx := basics.AppIndex(applicationID)
 	ledger := v2.Node.LedgerForAPI()
-
 	lastRound := ledger.Latest()
 
-	boxName, err := url.PathUnescape(boxName)
+	encodedBoxName := params.Name
```

Comment:
Is it true that if the param was url encoded, that it has already been unescaped here?

Reply:
@jannotti Yes, I think it's already been unescaped.  See rationale in https://github.com/algorand/go-algorand/pull/4183#discussion_r909784980.